cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java

```diff
@@ -207,6 +207,49 @@ public void testAdapterLogsPrev() throws Exception {
   }
 
   @Test
+  public void testWorkflowLogs() throws Exception {
+    testLogs("testTemplate1", "workflows", "testWorkflow1", MockLogReader.TEST_NAMESPACE, "testAdapter1");
+    try {
+      // in default namespace, we did not produce logs through adapters
+      testNext("testTemplate1", "workflows", "testWorkflow1", true, Constants.DEFAULT_NAMESPACE, "testAdapter1");
+      Assert.fail();
+    } catch (AssertionError e) {
```

Comment:
Great catch! This was a pattern throughout this test. Basically, we're expecting an ``AssertionError`` on line 214 above.  But throwing an ``Assert.fail()`` in the try defeats the purpose. I've fixed it in the next commit by moving this verification to the end of each test, and calling a return from the catch clause, then adding an ``Assert.fail()`` at the end of the test... also fixed it throughout this class.